### ISSUE TITLE
Druid book example does not compile

### ISSUE THREAD

**Issue by doinkythederp** [linebender/druid#2180]
The Druid book shows the [following example in chapter 3](https://linebender.org/druid/get_started.html):

```rust
use druid::{AppLauncher, WindowDesc, Widget, PlatformError};
use druid::widget::Label;

fn build_ui() -> impl Widget<()> {
    Label::new("Hello world")
}

fn main() -> Result<(), PlatformError> {
    AppLauncher::with_window(WindowDesc::new(build_ui())).launch(())?;
    Ok(())
}
```

The example does not compile, which can be fixed by not calling the `build_ui` function.

**Comment by werickdasilva** [comment]
Ops

Take the () from the build_ui function.
```rust
use druid::{AppLauncher, WindowDesc, Widget, PlatformError};
use druid::widget::Label;

fn build_ui() -> impl Widget<()> {
    Label::new("Hello world")
}

fn main() -> Result<(), PlatformError> {
    AppLauncher::with_window(WindowDesc::new(build_ui)).launch(())?;
    Ok(())
}
```
The error occurs because the new function receives an anonymous function as an argument

Oops sorry for my english, I'm still learning  :blush:

**Comment by werickdasilva** [comment]
> The Druid book shows the [following example in chapter 3](https://linebender.org/druid/get_started.html):
> 
> ```rust
> use druid::{AppLauncher, WindowDesc, Widget, PlatformError};
> use druid::widget::Label;
> 
> fn build_ui() -> impl Widget<()> {
>     Label::new("Hello world")
> }
> 
> fn main() -> Result<(), PlatformError> {
>     AppLauncher::with_window(WindowDesc::new(build_ui())).launch(())?;
>     Ok(())
> }
> ```
> 
> The example does not compile, which can be fixed by not calling the `build_ui` function.

**Comment by xStrom** [comment]
The book references the latest development version, which unfortunately does not match the latest released version (0.7.0). This will be fixed later this month with the release of 0.8.0. Thanks!
